event/src/app/api/events/category/route.ts

```diff
@@ -0,0 +1,24 @@
+import { CATEGORIES } from "@/lib/constants";
+import dbConnect from "@/lib/mongo";
+import Event from "@/models/Event";
+import { NextRequest, NextResponse } from "next/server";
+
+export async function GET(req: NextRequest) {
+  try {
+    await dbConnect()
+    let result: {} = {}
+    for (let i in CATEGORIES) {
+      const events = await Event.aggregate([{ $match: { "category": CATEGORIES[i] } }])
+        .addFields({ "participantCount": { "$size": '$participants' } })
+        .sort({ "participantCount": -1 })
+        .limit(3)
+      if (!events) {
+        continue
+      }
+      result = { ...result, [CATEGORIES[i]]: events }
+    }
```

Comment:
It doesn't seem like you're using `i` for anything else, so suggest using `of` instead of `in`.

```suggestion
    for (let category of CATEGORIES) {
      const events = await Event.aggregate([{ $match: { "category": category } }])
        .addFields({ "participantCount": { "$size": '$participants' } })
        .sort({ "participantCount": -1 })
        .limit(3)
      if (!events) {
        continue
      }
      result = { ...result, [category]: events }
    }
```